src/hooks/useViolations.ts

```diff
@@ -83,7 +83,11 @@ function useViolations(violations: TransactionViolation[], shouldShowOnlyViolati
 
             // someTagLevelsRequired has special logic becase data.errorIndexes is a bit unique in how it denotes the tag list that has the violation
             // tagListIndex can be 0 so we compare with undefined
-            if (currentViolations[0]?.name === CONST.VIOLATIONS.SOME_TAG_LEVELS_REQUIRED && data?.tagListIndex !== undefined && Array.isArray(currentViolations[0]?.data?.errorIndexes)) {
+            if (
+                currentViolations.at(0)?.name === CONST.VIOLATIONS.SOME_TAG_LEVELS_REQUIRED &&
+                data?.tagListIndex !== undefined &&
+                Array.isArray(currentViolations.at(0)?.data?.errorIndexes)
+            ) {
```

Comment:
You can move `const firstViolation = currentViolations.at(0);` above
```suggestion
            if (
                firstViolation?.name === CONST.VIOLATIONS.SOME_TAG_LEVELS_REQUIRED &&
                data?.tagListIndex !== undefined &&
                Array.isArray(firstViolation?.data?.errorIndexes)
            ) {
```